File: heartbeat_monitor/src/jsMain/resources/ThemeTogglePlugin.js
export function apply() {
    document.getElementById("topbar-right-container").prepend(
        also(document.createElement("button"), button => {
            button.type = "button";
            window.KanbanBro.event.addEventListener('themeChanged', () => {
                if (window.KanbanBro.theme === null) {
                    button.textContent = 'Theme: Auto';
                } else if (window.KanbanBro.theme === 'light') {
                    button.textContent = 'Theme: Light';
                } else if (window.KanbanBro.theme === 'dark') {
                    button.textContent = 'Theme: Dark';
                } else {
                    button.textContent = `Theme: ${window.KanbanBro.theme}`;
                }
            });
            button.addEventListener("click", () => {
                const currentTheme = window.KanbanBro.theme;
                setTheme(currentTheme === null ? "light" : currentTheme === "light" ? "dark" : null);
            });
        }),
    );
}

File: heartbeat_monitor/src/jsMain/kotlin/ThemeTogglePlugin.kt
package hello

import kotlinx.browser.document
import kotlinx.browser.window

object ThemeTogglePlugin : AbstractPlugin() {
    override suspend fun applyImpl() {
        UiContainers.topbarRightContainer.prepend(
            document.createElement("button").also { button ->
                button.asDynamic().type = "button"
                KanbanBro.event.addEventListener("themeChanged", { _ ->
                    button.textContent = when (val theme = KanbanBro.theme) {
                        null -> "Theme: Auto"
                        "light" -> "Theme: Light"
                        "dark" -> "Theme: Dark"
                        else -> "Theme: $theme"
                    }
                })
                button.addEventListener("click", { _ ->
                    val oldTheme = KanbanBro.theme
                    val newTheme = if (oldTheme == null) "light" else if (oldTheme == "light") "dark" else null
                    window.asDynamic().setTheme(newTheme)
                })
            },
        )
    }
}
